python/tvm/topi/cuda/tensorcore_alter_op.py

```diff
@@ -176,6 +176,12 @@ def _dense_legalize(attrs, inputs, arg_types):
 
     x_ = relay.nn.pad(x, pad_width=((0, dm), (0, dk))) if dm or dk else x
     y_ = relay.nn.pad(y, pad_width=((0, dn), (0, dk))) if dn or dk else y
+
+    # If units is explicitly specified, it is used to compute the output shape.
+    # We need to update units after padding to prevent a type error.
+    if attrs["units"] is not None:
+        new_attrs["units"] = N + dn
```

Comment:
I'm thinking about the semantic when `units` is specified but overwritten here. It looks like we are changing the expected output shape in this case, but it is fine if `units` is used basically internally. Is `units` mainly specified by users or other passes?

Reply:
`units` is specified by frontends. For example, PT and ONNX use it, while TF frontend doesn't. Actually I wonder why no one has hit this error before.

Reply:
Apparently, the dense tensorcore schedule is not used when batch size is 1. That's probably the reason this error is not frequent.

Reply:
I see make sense. Thanks for the explanation.